at_secondary/at_persistence_secondary_server/lib/src/compaction/commit_log_compaction_service.dart

```diff
@@ -24,7 +24,7 @@ class CommitLogCompactionService implements AtChangeEventListener {
             keysToCompactCount = keysToCompactCount + 1;
           }
           _commitLogEntriesMap.putIfAbsent(
-              commitEntry.atKey, () => CompactionSortedList());
+              commitEntry.atKey!, () => CompactionSortedList());
```

Comment:
Why is the bang needed here on commitEntry.atKey?

Reply:
I have type casted the toMap method in commitLogKeystore to Future<Map<int, CommitEntry>> from Future<Map<dyamic,dynamic>> 

Since the in the CommitEntry class, we have atkey defined as nullable - `final String? _atKey`, added a bang operator here. 
Further we will not get into a scenario of atKey going null since we have checks for them in at_client side and at_secondary_server